creme/stream/datasets/agrawal.py

```diff
@@ -0,0 +1,316 @@
+import numpy as np
+
+from . import base
+from creme.utils.skmultiflow_utils import check_random_state
+
+
+class AGRAWALGenerator(base.SyntheticDataset):
```

Comment:
How about calling this simply `Agrawal`? `AGRAWAL` isn't an acronym so I don't think that it should be capitalized. I also don't think that `Generator` brings anything meaningful.

If we want to make a distinction between finite datasets and infinite datasets (aka. generators) then I could add an `iter_generator` function in addition to `iter_dataset`. What do you think? :)

Reply:
> How about calling this simply Agrawal? AGRAWAL isn't an acronym so I don't think that it should be capitalized. I also don't think that Generator brings anything meaningful.

Agree and done.

> If we want to make a distinction between finite datasets and infinite datasets (aka. generators) then I could add an iter_generator function in addition to iter_dataset. What do you think? :)

+1 for `iter_generator`